lib/bolt/pal.rb

```diff
@@ -357,6 +359,39 @@ def get_task(task_name)
       Bolt::Task.from_task_signature(task)
     end
 
+    def list_plans_with_cache(filter_content: false)
+      # Don't filter content yet, so that if users update their plan filters
+      # we don't need to refresh the cache
+      plan_names = list_plans(filter_content: false).map(&:first)
+      plan_cache = if @project
+                     Bolt::Util.read_optional_json_file(@project.plan_cache_file, 'Plan cache file')
+                   else
+                     {}
+                   end
+      updated = false
+
+      plan_list = plan_names.each_with_object([]) do |plan_name, pl|
+        info = plan_cache[plan_name] || get_plan_info(plan_name, with_mtime: true)
+
+        # If the plan is a 'local' plan (in the project itself, or the
+        # modules/ directory) then verify it hasn't been updated since we
+        # cached it. If it has been updated, refresh the cache and use the
+        # new data.
+        if info['file'] &&
+           (File.mtime(info.dig('file', 'path')) <=> info.dig('file', 'mtime')) != 0
+          info = get_plan_info(plan_name, with_mtime: true).slice(*CACHE_KEYS)
```

Comment:
Am i reading this right... We are only caching whether the plan is private? Why not just cache all the parsed metadata on `get_plan_info` and use the cache for both `bolt plan show myplan` and `bolt plan show`?

Reply:
Uh, yeah, dang, I just missed the forest for the trees there. Will update, thanks Cas!

Reply:
Ah, ok, I see why I didn't at least - it's definitely going to make this already-big PR even bigger, and kind of feels like scope creep. I thought it'd be a good next-iteration improvement, but at this point I think it's reasonable to add that.